src/tokens/tokens.interfaces.ts

```diff
@@ -0,0 +1,201 @@
+// Copyright © 2022 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ApiProperty, OmitType } from '@nestjs/swagger';
+import { IsNotEmpty, IsOptional } from 'class-validator';
+
+// Ethconnect interfaces
+export interface EthConnectAsyncResponse {
+  sent: boolean;
+  id: string;
+}
+
+export interface EthConnectReturn {
+  output: string;
+}
+
+// REST API requests and responses
+export class AsyncResponse {
+  @ApiProperty()
+  id: string;
+}
+
+export enum TokenType {
+  FUNGIBLE = 'fungible',
+}
+
+const contractConfigDescription =
+  'Optional configuration info for the token pool. Reserved for future use.';
+const requestIdDescription =
+  'Optional ID to identify this request. Must be unique for every request. ' +
+  'If none is provided, one will be assigned and returned in the 202 response.';
+
+export class TokenPool {
+  @ApiProperty()
+  @IsNotEmpty()
+  name: string;
+
+  @ApiProperty()
+  @IsNotEmpty()
+  operator: string;
+
+  @ApiProperty()
+  @IsNotEmpty()
+  symbol: string;
```

Comment:
Are name and symbol absolutely required, or can they be blank?

We are also going to need to enhance FireFly to pass these down, and then enhance the ERC1155 connector to allow them as optional params. I think the connector will currently throw HTTP 400 if an unrecognized param is passed (can discuss that behavior as well, or if unrecognized params should be silently ignored).

Reply:
I believe they are required. Can discuss

Reply:
https://github.com/hyperledger/firefly/pull/413 passed here in Firefly